test/tests/LibStuffTest.cpp

```diff
@@ -594,4 +595,14 @@ struct LibStuff : tpunit::TestFixture {
         ASSERT_TRUE(SContains(string("asdf"), "a"));
         ASSERT_TRUE(SContains(string("asdf"), string("asd")));
     }
+
+    void testFirstOfNextMonth() {
+        string timeStamp = "2020-03-01";
+        string timeStamp2 = "2020-12-01";
+        string timeStamp3 = "2020-06-17";
```

Comment:
Add test for August or Sept (possibly octal months) and `"this is not a date"` for catching exceptions.